plugins/git/fugitive.nix

```diff
@@ -13,5 +13,12 @@ helpers.vim-plugin.mkVimPlugin config {
 
   maintainers = [ lib.maintainers.GaetanLepage ];
 
+  extraConfig = cfg: {
+    # mini.nvim and fugitive have duplicate doc tags
+    performance.combinePlugins.standalonePlugins = lib.mkIf (
```

Comment:
Having all plugins merged by default and marked standalone on a case-by-case basis makes me nervous.

Is it possible to add test cases for the `combinePlugins` option so we can know for sure that all plugins still work when it is enabled?

Reply:
If you mean to add a case with all plugins included, then I doubt it'll work. This option is not universal because of collisions. So it completely depends on what plugins user use. But in case of any collisions it won't build at all, so user will know instantly when he adds a new plugin to a config. User is expected to try to enable it, if it works - that's good, if not, than it need debugging. But some most often cases can be workarounded in nixvim beforehand, like I did with a couple of plugins here. So this performance option is not for all.

Reply:
I'm not sure we want to mark them as standalone, maybe we only want to add a warning in some way, so that the user choses what to do:
- Disable fugitive
- Disable mini
- Mark one of them as standalone

We could make this check in the performance module in case we add others in the future.

This may be harder to test though, because we need to check that a warning is indeed generated

Reply:
Warning in this case probably doesn't make much sense, because it just won't build at all when fugitive and mini both enabled. To be honest I'm also not a big fan about splitting them by default. In my config I just patched mini and removed mini.git's doc (because I don't use mini.git and it's doc is not needed for me), thus fixing duplicate doc tags. So in a user config it's easier to mark it as standalone than to unmark it back (is it even possible without mkForce that would also disable all other fixes?). It was as an example here on how it can be fixed in nixvim itself. But I agree to add only the most common fixes by default (like nvim-treesitter, because pretty much everyone is using it).

Reply:
I think there is a big difference between inter-plugin conflicts (where you have a choice of what to do) vs plugins with runtime stuff, where the fix is much obvious: make it standalone.

I know that it would error, but this does not mean that we could not generate a warning anyway, before having the conflict error I think. We may need to move some stuff around to ensure that we evaluate warnings before building neovim though, or we could have some special casing with `lib.warn` for this case

A warning is valuable because by experience people tend to open issues for such things (even if it marked experimental --')

Reply:
> I think there is a big difference between inter-plugin conflicts (where you have a choice of what to do) vs plugins with runtime stuff, where the fix is much obvious: make it standalone.

Could you rephrase it? I don't think I understand what's meant here.

> I know that it would error, but this does not mean that we could not generate a warning anyway, before having the conflict error I think.

But what would be a condition for the warning? If it's the same as here (mini.nvim + fugitive.vim both are enabled), than when user fixes the issue warning will stay... Add a check that none of the plugins are present in `standalonePlugins` to the condition? What if user fixed it some other way?

Reply:
> Could you rephrase it? I don't think I understand what's meant here.

I think there is a difference between the tree-sitter situation and fugitive/mini. tree-sitter has a single "obvious" fix, being make tree-sitter a standalonePlugin, as it conflicts with something in the core neovim derivation. fugitive/mini does not have a single fix, there could be a number of resolutions, and I don't think nixvim should do something

> But what would be a condition for the warning? If it's the same as here (mini.nvim + fugitive.vim both are enabled), than when user fixes the issue warning will stay... Add a check that none of the plugins are present in `standalonePlugins` to the condition? What if user fixed it some other way?

Ah indeed did not think how to silence the warning. Then the better solution would maybe to add a new page in the user guide listing all conflicts between plugins when enabling this option

Reply:
OK. Dropped `fugitive.vim` commit.